### PR TITLE
experimental search input: Refactor component and add telemetry

### PR DESCRIPTION
**Pull request by fkling**
The main purpose of this PR is to add event logging for option selection (see `ExperimentalSearchInput.tsx`). This needed to happen in the `client/web` components, but they had already been quite complex due to lazy loading extensions and suggestion sources.

To make this easier I created a new component,
`ExperimentalSearchInput`, ands its lazy loading equivalent, and moved all the extension and suggestion initialization logic in there. This simplifies the search home input and navbar input components and gets rid of the complex extension lazy loading logic (at the expense of having a more complex component interface).

## Test plan

- Loaded the search home page and the search results page. The experimental search input is rendered correctly.
- Enable telemetry logging via `slocalStorage.setItem('eventLogDebug', true)` and select entries via (`Shift`+)`Return` or the mouse logs telemetry events.
- Toggling the menu button enables/disables the experimental search input and logs the corresponding events.

## App preview:

- [Web](https://sg-web-fkling-search-input-metrics.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/search/input/ExperimentalSearchInput.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+const eventNameMap: Record<Action['type'], string> = {
+    completion: 'Add',
+    goto: 'GoTo',
+    command: 'Command',
+}
+
+export interface ExperimentalSearchInputProps
+    extends Omit<CodeMirrorQueryInputWrapperProps, 'suggestionSource' | 'extensions' | 'placeholder'>,
+        TelemetryProps,
```
Comment (fkling) [inline client/web/src/search/input/ExperimentalSearchInput.tsx:26]: I don't know if there is a better way these days to access the logger... could I just import it directly?

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.04% (-1.13 kb) | -0.10% (-14.70 kb) 🔽 | -0.12% (-13.57 kb) 🔽 | -0.54% (-4) 🔽 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/fkling/search-input-metrics/compare-report.html;tab=live_object) for a full comparison between the commits a971051 and ddf69c6 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by rrhyne — APPROVED** [review]
Metrics events look great.

File: client/web/src/search/input/ExperimentalSearchInput.tsx (hunk 1 above)
Comment (philipp-spiess) [reply client/web/src/search/input/ExperimentalSearchInput.tsx:26]: You can directly import it, yeah, and we do in a lot of places 😅  Not sure if this is the pattern we want to use though, maybe @valerybugakov has some ideas?

For now this is also fine though, we also use this pattern in a lot of places!

**Review by philipp-spiess — APPROVED** [review]
Nice cleanups!

**Pull request merged**